components/ledger/pkg/machine/vm/program/resource.go

```diff
@@ -8,22 +8,28 @@ import (
 
 type Resource interface {
 	GetType() core.Type
+	GetMonetaryAsset() *core.Address
```

Comment:
Adding those methods on the Resource interface feels like a code smell.
Since you have the type with GetType(), maybe you could simply cast the interface when needed?

Reply:
I agree that this is not ideal, but I can't cast the interface with only the type, because a `core.TypeMonetary` can be of concrete type `program.Monetary` or `program.VariableAccountBalance`.